### PR TITLE
Add oauth scope support

### PR DESCRIPTION
**Pull request by jason-chan-lf**
Add oauth scope support

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
+### Features
 - **[BREAKING]** For every API, the parameters are now encapsulated into a parameter class.
 - **[BREAKING]** Enum values are now following Java naming convention.
 - **[BREAKING]** Every async API are replaced with normal blocking API.
-- The ApiClient class is now abstract.
 - Improved documentation.
+- Added retry implementation when making an API call which retries when:
+  - 401 status code response is being returned
+  - Idempotent http request
+- Added an option when constructing the `RepositoryApiClient` to specify the request scope(s) for the access token.
```
Comment (jason-chan-lf) [inline CHANGELOG.md]: I still need to update the api client core to a new preview package

Comment (jason-chan-lf) [reply CHANGELOG.md]: updated the client core package version

Comment (terrencejames-lf) [inline CHANGELOG.md]: requested?

Comment (jason-chan-lf) [reply CHANGELOG.md]: updated

**Review by terrencejames-lf — APPROVED** [review]

**Pull request merged**